atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/TimestampClient.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.jepsen;
+
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Lists;
+import com.jayway.awaitility.Awaitility;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.timestamp.TimestampService;
+
+public final class TimestampClient {
+    private static final int PORT = 8080;
+    private static final String NAMESPACE = "test";
+    private static final int TIMEOUT_SECONDS = 60;
+
+    private TimestampClient() {
+    }
+
+    public static TimestampService create(List<String> hosts) {
+        List<String> endpointUris = hostnamesToEndpointUris(hosts);
+        return createFromUris(endpointUris);
+    }
+
+    public static TimestampService randomizeHostsAndCreate(List<String> hosts) {
+        List<String> randomizedHosts = new ArrayList<>(hosts);
+        Collections.shuffle(randomizedHosts);
+        return create(randomizedHosts);
+    }
+
+    public static void waitUntilHostReady(String host) {
+        Awaitility.await()
+                .atMost(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .until(() -> hostIsListening(host));
+    }
+
+    public static void waitUntilTimestampClusterReady(List<String> hosts) {
+        Awaitility.await()
+                .atMost(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .until(() -> clusterReturnsTimestamp(hosts));
+    }
+
+    private static boolean clusterReturnsTimestamp(List<String> hosts) {
```

Comment:
This is used in a polling function that is waiting until the timelock server is ready. The first time it is called, the timelock server is still starting up. My motivation was that it would be better to keep re-creating the service than to create one before the timelock server was ready and keep using it, in case it got into a bad state somehow.

Do you still think it's better just to create one?